misk-service/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -19,22 +19,21 @@ internal class CoordinatedService(
         created.checkNew("$created must be NEW for it to be coordinated")
         created.addListener(
           object : Listener() {
+
+            val outerService = this@CoordinatedService
+
             override fun running() {
-              synchronized(this) {
```

Comment:
Don't think these `synchronized` locks here actually achieve anything?

Reply:
These synchronized blocks have caused us problems before (e.g. in #1958, with partial reversion in #1976). If the notifyX methods all lock internally, then yeah, let's not lock here.